engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -654,10 +663,7 @@ namespace dmGameSystem
             DM_PROPERTY_ADD_U32(rmtp_ModelVertexCount, buffers->m_VertexCount);
             DM_PROPERTY_ADD_U32(rmtp_ModelVertexSize, buffers->m_VertexCount * sizeof(dmRig::RigModelVertex));
 
-            for(uint32_t i = 0; i < MAX_TEXTURE_COUNT; ++i)
-            {
-                ro.m_Textures[i] = GetMaterialTexture(component, material_index, i);
-            }
+            FillTextures(&ro, component, material_index);
```

Comment:
Pro: We don't iterate over MAX_TEXTURE_COUNT items anymore.
Con:  It is a more complex loop.
If it becomes a problem, we might want to cache this info in the model resource or component, and perhaps use hashtables instead.

Reply:
Also, another way would be to defer that sorting/lookup to the renderer.
So instead, we'd pass the sampler/texture pairs to the renderer via the RenderObject.
It might simplify all the workings around this, at the cost of the renderobject growing a little bit